src/app/sell/sell.class.ts

```diff
@@ -0,0 +1,50 @@
+import { ISell } from './sell.interface';
+import { Sell as SellType } from './sell.type';
+import { Cake } from '../cake/cake.type';
+
+export class Sell implements ISell {
+	id: number
+  customerName: string
+  customerPhoneNumber: string
+  customerEmail: string
+  totalAmount: number
+  cake: Omit<Cake, 'id' | 'stock' | 'status'> & { quantity: number }  
+	
+
+	constructor(
```

Comment:
It would be a good idea to inject your services, if you do it this way you would be able to inject a mock of these services for testing purposes.

```typescript
export type BreadInjection = {
  breadService?: BreadService;
}
```
```typescript
  constructor(injection: BreadInjection = {}) {
    this.breadService = injection?.breadService || new BreadService(dbMain);
  }
```